### PR TITLE
Add request received and elapsed time info to logger

### PR DESCRIPTION
**Pull request by r-ash**
This PR will
* Add `request_received` date time in UTC and elapsed time to `postserialize` log message

I think this will be generally useful info to have but I specifically want to see this for hintr to confirm that the load issues we are having are because hintr is responding slowly to some requests.

I added a custom formatter here instead of using the default difftime formatter because I want to see the times formatting in ms. As I can see default formatters lowest unit is seconds.

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -270,3 +270,34 @@ list_call <- function(fn, x) {
...
+    } else {
+      "hours"
+    }
+  }
+  time <- switch(units,
+                 ms = z * 1000,
+                 secs = z,
+                 mins = z / 60,
+                 hours = z / 3600)
+  paste(round(time, 2), units)
```
Comment (richfitz) [inline R/utils.R]:
```suggestion
  units_difftime <- if (units == "ms") "secs" else units
  z <- as.numeric(time1 - time2, units_difftime)
  if (units == "ms") {
    z <- z / 1000
  }
  sprintf("%.2f %s", z, units)
```

Somewhat simpler implementation. If you want to drop the trailing zero then process the z first to make it

```r
  paste(formatC(z, digits = 2, format = "f", drop0trailing = TRUE), units)
```

File: tests/testthat/test-utils.R [hunk 2]
```diff
@@ -142,3 +142,16 @@ test_that("try and find a free port", {
     fixed = TRUE)
   mockery::expect_called(mock_check_port, 5)
 })
+
+
+test_that("can format difftime nicely", {
+  now <- Sys.time()
```
Comment (richfitz) [inline tests/testthat/test-utils.R]: consider making the test deterministic by doing:

```suggestion
  now <- ISOdate(2022, 11, 11, 10, 49, 33.123)
```

File: R/utils.R [hunk 3]
```diff
@@ -270,3 +270,34 @@ list_call <- function(fn, x) {
...
+now_utc <- function() {
+  as.POSIXct(Sys.time(), tz = "UTC")
+}
+
+## Modified version of difftime to show timings
+## in ms and to a precision
+format_difftime <- function(time1, time2) {
+  z <- unclass(time1) - unclass(time2)
+  attr(z, "tzone") <- NULL
+  units <- if (all(is.na(z))) {
```
Comment (richfitz) [inline R/utils.R]: why was the all() here, or min(abs(z))? I don't see when this is used with other than scalar args?

**Review by richfitz — CHANGES_REQUESTED** [review]

File: R/utils.R (hunk 3 above)
Comment (r-ash) [reply R/utils.R]: just copied from `difftime` impl but yeah I think you're right it is not needed here

**Review by richfitz — APPROVED** [review]

**Pull request merged**